packages/odf/components/create-storage-system/create-storage-system-steps/security-and-network-step/nic.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { NetworkType } from '@odf/core/types';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { NICSelectComponent } from './nic';
+
+jest.mock('@odf/shared', () => ({
+  useCustomTranslation: () => ({ t: (key: string) => key }),
+}));
+
+describe('NICSelectComponent', () => {
+  const setPublicCIDR = jest.fn();
+  const setCephCIDR = jest.fn();
+  const setNetworkType = jest.fn();
+
+  const renderComponent = (networkType: NetworkType) =>
+    render(
+      <NICSelectComponent
+        cephClusterCIDR="192.168.100.0/24"
+        cephPublicCIDR="192.168.0.0/32"
+        setPublicCIDR={setPublicCIDR}
+        setCephCIDR={setCephCIDR}
+        networkType={networkType}
+        setNetworkType={setNetworkType}
+      />
+    );
+
+  it('should render the component with NIC network type', () => {
+    renderComponent(NetworkType.NIC);
+    expect(
+      screen.getByLabelText('Isolate network using NIC Operators')
+    ).toBeChecked();
+    expect(screen.getByText('Ceph Cluster CIDR')).toBeInTheDocument();
+    expect(screen.getByText('Ceph Public CIDR')).toBeInTheDocument();
+  });
+
+  it('should render the component with HOST network type', () => {
+    renderComponent(NetworkType.HOST);
+    expect(
+      screen.getByLabelText('Isolate network using NIC Operators')
+    ).not.toBeChecked();
+    expect(screen.queryByText('Ceph Cluster CIDR')).not.toBeInTheDocument();
+    expect(screen.queryByText('Ceph Public CIDR')).not.toBeInTheDocument();
+  });
+
+  it('should call setNetworkType when checkbox is clicked', () => {
+    renderComponent(NetworkType.HOST);
+    fireEvent.click(
```

Comment:
Please use `userEvent` instead:
```suggestion
import userEvent from '@testing-library/user-event';
[...]
    const user = userEvent.setup();
    await user.click(...
```